### PR TITLE
Add imgui 1.74

### PR DESCRIPTION
**Pull request by Hopobcn**
Specify library name and version:  **imgui/1.74** https://github.com/ocornut/imgui

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Recipy based on https://github.com/bincrafters/conan-imgui
Tested on a Ubuntu 18.04. Also tested modifying conan-examples/libs/dear-imgui (had to change the imports from `misc` to `res`)

### CODE REVIEW

File: recipes/imgui/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+    
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.verbose = True
```
Comment (uilianries) [inline recipes/imgui/all/conanfile.py]:
```suggestion
```

By default, we don't enable verbose output.

File: recipes/imgui/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from conans import ConanFile, CMake, tools
+import os
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.verbose = True
```
Comment (Hopobcn) [inline recipes/imgui/all/test_package/conanfile.py]: I guess I should also remove this verbose.

Comment (uilianries) [reply recipes/imgui/all/test_package/conanfile.py]: go ahead!

File: recipes/imgui/all/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,48 @@
+cmake_minimum_required(VERSION 2.8.11)
+project(imgui CXX)
+
+include(conanbuildinfo.cmake)
+conan_basic_setup()
+
+if (WIN32 AND MSVC AND BUILD_SHARED_LIBS)
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```
Comment (danimtb) [inline recipes/imgui/all/CMakeLists.txt]: This requires cmake 3.4, I'd suggest increasing the minimum version

**Comment by Hopobcn** [comment]
> I would like to clarify the usage of the CMakeLists.txt file, it it needed to build the bindings?

imgui does not provide any associated build script. It delegates the task of compiling the library to their users.
From the readme:

> The core of Dear ImGui is self-contained within a few platform-agnostic files which you can easily compile in your application/engine. They are all the files in the root folder of the repository (imgui.cpp, imgui.h, imgui_demo.cpp, imgui_draw.cpp etc.).
> 
> No specific build process is required. You can add the .cpp files to your existing project.

> You will need a backend to integrate Dear ImGui in your app. The backend passes mouse/keyboard/gamepad inputs and variety of settings to Dear ImGui, and is in charge of rendering the resulting vertices.

So  the `CMakeLists.txt` is necessary to compile the cpp files (`imgui.cpp`, `imgui_demo.cpp`, `imgui_draw.cpp` & `imgui_widgets.cpp`) into a `imgui` library.

Then the platform backends are provided to the user inside the `res/bindings` folder. It's the consumer responsibility to cherry-pick which platform binding to use (via `imports`):

```
[imports]
./res/bindings, imgui_impl_glfw.cpp -> ../bindings
./res/bindings, imgui_impl_opengl3.cpp -> ../bindings
./res/bindings, imgui_impl_glfw.h -> ../bindings
./res/bindings, imgui_impl_opengl3.h -> ../bindings
```

I don't know if this is up to conan-center-index standards but at least this is how it is done in the bincrafters recipy.

**Pull request merged**